spark/v2.4/spark/src/test/java/org/apache/iceberg/spark/source/TestIcebergSourceTablesBase.java

```diff
@@ -1415,6 +1415,89 @@ public void testRemoveOrphanFilesActionSupport() throws InterruptedException {
     Assert.assertEquals("Rows must match", records, actualRecords);
   }
 
+  @Test
+  public void testOverwritePartition() {
+    TableIdentifier tableIdentifier = TableIdentifier.of("db", "entries_test");
+    Table table = createTable(tableIdentifier, SCHEMA, PartitionSpec.builderFor(SCHEMA).identity("id").build());
+
+    Dataset<Row> df1 = spark.createDataFrame(Lists.newArrayList(new SimpleRecord(1, "a")), SimpleRecord.class);
+    df1.select("id", "data").write()
+            .format("iceberg")
+            .mode("append")
+            .save(loadLocation(tableIdentifier));
+
+    List<SimpleRecord> result = spark.read()
+            .format("iceberg")
+            .load(loadLocation(tableIdentifier))
+            .as(Encoders.bean(SimpleRecord.class))
+            .collectAsList();
+    Assert.assertEquals(result, Lists.newArrayList(new SimpleRecord(1, "a")));
+
+    // Now, we try to overwrite the partition id=1
+    Dataset<Row> df2 = spark.createDataFrame(Lists.newArrayList(new SimpleRecord(1, "b")), SimpleRecord.class);
+    df2.select("id", "data").write()
+            .format("iceberg")
+            .mode("overwrite")
+            .option("overwrite-partitions", "id=1")
+            .save(loadLocation(tableIdentifier));
+
+    result = spark.read()
+            .format("iceberg")
+            .load(loadLocation(tableIdentifier))
+            .as(Encoders.bean(SimpleRecord.class))
+            .collectAsList();
```

Comment:
Can you fix the formatting of this file? Continuation indents should be 4 spaces, not 8.